qiskit/transpiler/passes/routing/layout_transformation.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Map (with minimum effort) a DAGCircuit onto a `coupling_map` adding swap gates."""
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.routing import util
+from qiskit.transpiler.routing.general import ApproximateTokenSwapper
+
+
+class LayoutTransformation(TransformationPass):
+    """ Adds a Swap circuit for a given (partial) permutation to the circuit.
+
+    This circuit is found by a 4-approximation algorithm for Token Swapping.
+    More details are available in the routing code.
+    """
+
+    def __init__(self, coupling_map, initial_layout, final_layout, trials=4):
```

Comment:
Since ApproximateTokenSwapper is a randomized algorithm, we'd better to have `seed` in the arguments so that users can fix the seed and reproduce the transpilation results when they want. [StochasticSwap](https://github.com/Qiskit/qiskit-terra/blob/d5f67b0cc4362491010a1a606bb41b905ec8ad41/qiskit/transpiler/passes/routing/stochastic_swap.py#L53) is an good example demonstrating how to handle seeds in a pass.